app/forms/AddressFormProvider.scala

```diff
@@ -41,38 +42,39 @@ class AddressFormProvider @Inject() extends Mappings {
     Form(mapping(
       "property" -> optional(text()
         .verifying(maxLength(propertyMax, "address.property.error.length"))
-        .verifying(regexp(ALPHANUMERIC_REGEX, "address.property.error.characters"))
+        .verifying(regexp(ALPHANUMERIC_REGEX, "address.property.error.character"))
         .verifying(regexpUnlessEmpty(XSS_REGEX, "address.property.error.invalid"))),
 
       "street" -> text("address.street.error.required")
         .verifying(maxLength(streetMax, "address.street.error.length"))
-        .verifying(regexp(ALPHANUMERIC_REGEX, "address.street.error.characters"))
+        .verifying(regexp(ALPHANUMERIC_REGEX, "address.street.error.character"))
         .verifying(regexpUnlessEmpty(XSS_REGEX, "address.street.error.invalid")),
 
       "town" -> text("address.town.error.required")
         .verifying(maxLength(townMax, "address.town.error.length"))
-        .verifying(regexp(ALPHANUMERIC_REGEX, "address.town.error.characters"))
+        .verifying(regexp(ALPHANUMERIC_REGEX, "address.town.error.character"))
         .verifying(regexpUnlessEmpty(XSS_REGEX, "address.town.error.invalid")),
 
       "postcode" -> text("address.postcode.error.required")
         .verifying(maxLength(postcodeMax, "address.postcode.error.length"))
-        .verifying(regexp(ALPHANUMERIC_REGEX, "address.postcode.error.characters"))
+        .verifying(regexp(ALPHANUMERIC_REGEX, "address.postcode.error.character"))
         .verifying(regexpUnlessEmpty(XSS_REGEX, "address.postcode.error.invalid"))
         .verifying(getExtraPostcodeValidationForPage(page): _*)
     )(UserAddress.apply)(UserAddress.unapply)
     )
 
   private def getExtraPostcodeValidationForPage(page: Page)(implicit request: DataRequest[_]): Seq[Constraint[String]] = {
 
-    def niPostcode(isNi: Boolean): Seq[Constraint[String]] =
-      if(isNi) Seq(startsWith(XI_POSTCODE, "address.postcode.error.mustStartWithBT")) else {
-        Seq(doesNotStartWith(XI_POSTCODE, "address.postcode.error.mustNotStartWithBT"))
+    def niPostcode(isNi: Boolean, page: Page): Seq[Constraint[String]] =
+      if(isNi) Seq(startsWith(XI_POSTCODE, s"address.postcode.error.$page.mustStartWithBT")) else {
+        Seq(doesNotStartWith(XI_POSTCODE, s"address.postcode.error.$page.mustNotStartWithBT"))
       }
 
     page match {
-      case ConsignorAddressPage => niPostcode(request.isNorthernIrelandErn)
+      case ConsignorAddressPage => niPostcode(request.isNorthernIrelandErn, page)
+      case ConsigneeAddressPage => niPostcode(request.isNorthernIrelandErn, page)
```

Comment:
Sorry, only just spotted this now. `request.isNorthernIrelandErn` is the consignors ERN not what the user entered on the consignee excise page (i.e., CAM-NEE01)